cscs-checks/prgenv/environ_check.py

```diff
@@ -41,3 +42,44 @@ def __init__(self):
     @sn.sanity_function
     def env_module_patt(self):
         return r'^%s' % self.current_environ.name
+
+
+@rfm.parameterized_test(['cray-fftw'], ['cray-hdf5'], ['cray-hdf5-parallel'],
+                        ['cray-libsci'], ['cray-netcdf'],
+                        ['cray-netcdf-hdf5parallel'], ['cray-petsc'],
+                        ['cray-petsc-64'], ['cray-petsc-complex'],
+                        ['cray-petsc-complex-64'], ['cray-python'],
+                        ['cray-R'], ['cray-tpsl'], ['cray-tpsl-64'],
+                        ['cudatoolkit'], ['gcc'], ['papi'], ['pmi'])
+class CrayVariablesCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self, variant):
+        self.variant = variant
+        self.descr = 'Check for standard Cray variables'
+        self.valid_systems = ['daint:login', 'dom:login']
+        self.valid_prog_environs = ['builtin']
+
+        self.executable = 'module'
+        self.executable_opts = ['show', variant]
+
+        mod_name = variant.upper().replace('-', '_')
+
+        self.sanity_patterns = sn.all([
+            sn.assert_found(f'{mod_name}_PREFIX', self.stderr),
+            sn.assert_found(f'{mod_name}_VERSION', self.stderr)
+        ])
+
+        self.maintainers = ['EK', 'VH']
+        self.tags = {'production', 'craype'}
+
+    @rfm.run_before('sanity')
+    def cdt_workaround(self):
+        cdt = osext.cray_cdt_version()
+        if not cdt:
+            return
+
+        # these modules should be fixed in later releases
+        if (cdt <= '20.10' and
+            self.variant in ['cray-petsc-complex',
+                             'cray-petsc-complex-64',
+                             'cudatoolkit', 'gcc']):
```

Comment:
You could explicitly empty the `valid_systems` for these modules and for CDT <= 20.10 directly in the `__init__()`. This way these tests will not even be run if the cdt is not the right one.